src/libxively/mqtt/codec/xi_mqtt_parser.c

```diff
@@ -489,6 +493,7 @@ xi_state_t xi_mqtt_parser_execute( xi_mqtt_parser_t* parser,
     {
         xi_debug_logger( "XI_MQTT_ERROR_PARSER_INVALID_MESSAGE_ID" );
         parser->error = XI_MQTT_ERROR_PARSER_INVALID_MESSAGE_ID;
+        XI_CR_RESET( parser->read_cs );
```

Comment:
Sort of - the problem with local static variable that kept the coroutine state was that in between fuzz tests calls the static variable couldn't be zeroed - so the elements of the parser were "continuing" execution from the remembered state. This could also be a potential issue with restarting the the xively client library while still receiving a msg.